src/xcode/ENA/ENASecurity/Sources/ENASecurity/ECKeyPairGeneration.swift

```diff
@@ -0,0 +1,108 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+public enum ECKeyPairGenerationError: Error {
+    case privateKeyGenerationError(String?) // if key cannot be used for AES
+    case publicKeyGenerationFailed // if key cannot be used for AES
+    case dataGenerationFromKeyFailed(String?) // if key cannot be used for AES
+}
+
+public struct ECKeyPairGeneration {
+    
+    // MARK: - Public
+
+    func generateECPair(with name: String? = nil) -> Result<ECKeyPair, ECKeyPairGenerationError> {
+        let secureKey = self.generatePrivateKey(with: name)
+        guard let privateKey = secureKey.0 else {
```

Comment:
Not mandatory, just for information: If the other functions would also return a Result type, you could just concatenate all the calls with flatmap. This way this func will only be a couple of lines. (https://useyourloaf.com/blog/using-swift-result-and-flatmap/)

Reply:
they are not returning a Result type, they return an error message or no error and based on the function i return the needed enum